### ISSUE TITLE
RoadMap. Features list. 

### ISSUE THREAD

**Issue by podarok** [drupal-ukraine/advanced_forum#2]
- [x] Prepare environment #3 
  - [x] VM #3 
  - [x] D7 #3 
  -  [x] D8 #5 
  -  [x] Contribs #4 
  -  [x] Initial Advanced Forum module #6 
- [x] For ability to build proper Architecture we need to investigate features list that already is in D7 version.
- [ ] After getting full list of features - we need to draw a line over features that covered by core and stable contribs for not copy their functionality, but just reuse it within @drupal-ukraine/advanced-forum project
- [ ] for the list of unique and needed features we need to create Architecture and to approach them as a project for D8

**Comment by alexschedrov** [comment]
List of features https://www.drupal.org/node/841056

**Comment by infopicard** [comment]
any progress on this project?

**Comment by podarok** [comment]
@infopicard we are expecting human resources issues, so feel free to continue porting

**Comment by infopicard** [comment]
Thank you for your trust, but I'm not a developer, I can only help when
tests starts.

2016-02-24 10:31 GMT+01:00 Andrii Podanenko notifications@github.com:

**Comment by podarok** [comment]
@infopicard thank you for letting us know
We will send you issues if there are needs for testing something. Can you be more descriptive what features you need at first for advanced forum?

**Comment by richardmbell** [comment]
I am, primarily, looking for the capability to subscribe to a forum/thread so that users will get an email notification when there are new postings.

**Comment by mamair** [comment]
A very basic feature is to set a forum "private" and to select hich users/roles have access to a forum. users with a specific role should get the right to make their forum private.

There was a module forum access, I think, this should be part of advanced forum!

https://www.drupal.org/project/forum_access

I can also help to test and to translate into german

Update: Found a promising modul that would solve my request, but it still didnt work on my site (no error, but no denial of access for users without the right permissions): Permission Term https://www.drupal.org/project/permissions_by_term

@richardmbell Some features could be done by other modules. Maybe the subscription modul can do what you are looking for: https://www.drupal.org/project/subscriptions

**Comment by lostcarpark** [comment]
@mamair I agree forum access is required for D8 forums, but is there any reason it should be incorporated into Advanced Forum rather than handled by separate modules?

**Comment by mamair** [comment]
@lostcarpark If there is a working solution: no. Weeks ago when I was looking for one, i didnt find a workung solution, taxonomy access didnt work :-(
A reference to related modules on advance forum or forum page would be helpfull.

**Comment by LowellMontgomery** [comment]
The [ForumAccess](https://www.drupal.org/project/forum_access) module does not yet have a D8 version, although there is [a ticket open for porting of it](https://www.drupal.org/node/2774907) (maintainers could use support). It depends on [ACL (Access Control Lists)](https://www.drupal.org/project/acl), which has an alpha1 for D8; the ACL maintainers are asking for support to finish the port.

So that is the current status of what blocks using the standard external modules. Some degree of "user access control" functionality might be helpful to build directly into this module if those are not yet stable and/or if helping bring them to a stable release might cause too much delay. Building such functionality directly into this module might not be ideal, but it would effectively circumvent the "blockers".

**Comment by 32i** [comment]
We'd like to help with migrating forum access to D8.
I'll create separate ticket for this.

**Comment by arnoldbird** [comment]
Some might find this useful:  https://www.drupal.org/project/gforum
I'm the author.  Feedback is welcome.  I don't know how it compares with some of the other solutions mentioned here.  I've never used Advanced Forum.

**Comment by BSimon47** [comment]
The last post here is from January 2017...
I am so surprised that this valuable addition to the Drupal Community has seemingly been dropped!

The Core Forum for Drupal is so far behind in it's functionality and styling and the need for a quality Community Forum solution seems to me to be fundamental.

I looked at a solution via Disqus, but although that would seemingly be a good answer, it appears that they have moved from Forum to strictly Content Commenting Threads.

Bottom line, can ANYONE project a timeline for this module to be ready for D8?

Thanks

**Comment by Dragonwalker** [comment]
I really want to migrate to D8. Our site is very basic at this time and I had set it up a while ago under Drupal 7.
The forum is the only part completed and is exactly as I wish it to be.
Can I go ahead and migrate anyway?

Thanks.

**Comment by dnebrich** [comment]
I started w/ trying to get a forum module going.  In the end, I self-hosted discourse.  Very pleased with the result.  Used the cas_server module to support SSO.  I recommend just integrating with an external solution.  With discourse, you can self-host and own all of the data.

Had similar experiences with comments and chat.  For comments, went with coral project talk.  Here, getting the SSO integration going was a little trickier, but in the end it was worth it.  For chat, I integrated rocket.chat.
